### PR TITLE
Move OTLP spec to github.com/open-telemetry/opentelemetry-proto

### PR DESCRIPTION
**Pull request by tigrannajaryan**
Contributes to https://github.com/open-telemetry/opentelemetry-specification/issues/3415

This PR copies the following files exactly as they are without any modifications:

```
specification/protocol/README.md
specification/protocol/design-goals.md
specification/protocol/otlp.md
specification/protocol/requirements.md
```

The PR contains all historic commits that contributed to these files. We need to merge this PR without squashing!

Next steps after this PR is merged:
- Fix links.
- Copy markdown link check and linting make targets and github actions from spec repo.
- Remove content from the files in open-telemetry/opentelemetry-specification repo, add short messages in each file referring to the new location, without deleting the files, so that old references do not become 404.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
> * ❌ The commit ([1c725e7](https://github.com/open-telemetry/opentelemetry-proto/commit/1c725e7414b9d1d3501eab7498a4b2fb9108087f), [169ed62](https://github.com/open-telemetry/opentelemetry-proto/commit/169ed620885cebcc073be4cb6ba88b7a36e8f72c), [169ed62](https://github.com/open-telemetry/opentelemetry-proto/commit/169ed620885cebcc073be4cb6ba88b7a36e8f72c), [f2fb104](https://github.com/open-telemetry/opentelemetry-proto/commit/f2fb104e55c4c55693fcaa770df0885dffa2190d), [9db4949](https://github.com/open-telemetry/opentelemetry-proto/commit/9db49492755789d15d344046afac3282c42ef22e)). This user is missing the User's ID, preventing the EasyCLA check. [Consult GitHub Help](https://help.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user) to resolve.For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

EasyCLA is wrong. I verified all commits and they are from members who I know had signed the CLA.

**Comment by yurishkuro** [comment]
> This PR copies the following files exactly as they are without any modifications:

can you illustrate how you copied files with the commits history?

**Comment by tigrannajaryan** [comment]
> > This PR copies the following files exactly as they are without any modifications:
> 
> can you illustrate how you copied files with the commits history?

I produced a patch using git log that only includes commits that touched the relevant files:
```
git log --pretty=email --patch-with-stat --reverse --full-index --binary -m --first-parent -- specification/protocol/img specification/protocol/design-goals.md specification/protocol/otlp.md specification/protocol/README.md specification/protocol/requirements.md > patch
```

Then applied the patch in the proto repo:
```
git am --committer-date-is-author-date < patch 
```

The resulting commit hashes are obviously different, since the patch is applied on top of the current HEAD. I plan to merge this PR without squashing to keep this commit history.

If there is a better way to do this, I am happy to try it.

**Comment by jsuereth** [comment]
@tigrannajaryan Clever!  I've use git filter-branch before to extract out components preserving history, but I never had to merge it with existing history. I think what you did here makes the most sense!

**Review by jsuereth — APPROVED** [review]

**Review by jmacd — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

**Review by reyang — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
> * ❌ The commit ([1c725e7](https://github.com/open-telemetry/opentelemetry-proto/commit/1c725e7414b9d1d3501eab7498a4b2fb9108087f), [169ed62](https://github.com/open-telemetry/opentelemetry-proto/commit/169ed620885cebcc073be4cb6ba88b7a36e8f72c), [169ed62](https://github.com/open-telemetry/opentelemetry-proto/commit/169ed620885cebcc073be4cb6ba88b7a36e8f72c), [f2fb104](https://github.com/open-telemetry/opentelemetry-proto/commit/f2fb104e55c4c55693fcaa770df0885dffa2190d), [9db4949](https://github.com/open-telemetry/opentelemetry-proto/commit/9db49492755789d15d344046afac3282c42ef22e)). This user is missing the User's ID, preventing the EasyCLA check. [Consult GitHub Help](https://help.github.com/en/github/committing-changes-to-your-project/why-are-my-commits-linked-to-the-wrong-user) to resolve.For further assistance with EasyCLA, [please submit a support request ticket](https://jira.linuxfoundation.org/servicedesk/customer/portal/4).

@open-telemetry/governance-committee are you OK with overriding the EasyCLA on this PR? From what I see this is a false alarm. The commits in question are from @SergeyKanzhelev @arminru @bogdandrutu and @yurishkuro who all signed the CLA.

**Review by jack-berg — APPROVED** [review]

File: specification/protocol/design-goals.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Design Goals for OpenTelemetry Wire Protocol
```
Comment (jack-berg) [inline specification/protocol/design-goals.md:1]: Can we put these documents in `/specification` instead of `/specification/protocol`?

Comment (tigrannajaryan) [reply specification/protocol/design-goals.md:1]: Yes, I we can move after we merge this PR. I wanted to keep the patch exactly as it was produced by the git commands I used for others to verify.

There is a few other cleanup steps needed after the PR is merged (update links, add link checking, etc).

**Review by arminru — APPROVED** [review]

**Comment by arminru** [comment]
/easycla

(trying this as per https://github.com/open-telemetry/community/issues/1445#issuecomment-1518148952, cc @trask)

**Comment by arminru** [comment]
^ @tigrannajaryan CLA is passing now ✅ 

image

**Comment by tigrannajaryan** [comment]
Thanks @arminru @trask for the EasyCLA fix!

**Comment by tigrannajaryan** [comment]
I am going to disable "squash" option temporarily to merge this.

**Comment by tigrannajaryan** [comment]
> I am going to disable "squash" option temporarily to merge this.

Correction. I enabled "Rebase and Merge" option. If I merge this now, we will have 35 new commits. This will NOT create a merge commit. Anyone thinks we should have a merge commit instead?

**Comment by reyang** [comment]
> Correction. I enabled "Rebase and Merge" option. If I merge this now, we will have 35 new commits.

Sounds reasonable to me 👍

**Pull request merged**